feat: improve webapp, improve mobile player, improve design ✨

IEduStu opened the pull request:
Stremio is awesome and I use it a lot, so I wanted to add the missing features that I think would make its PWA on mobile (and on iOS and iPadOS in particular) first-class.

### Changes
* Improved the look and feel on mobile devices
* Improved the look and feel of the mobile PWA app - it now feels much more like a native app on iOS
* Improved the design of the mobile app and desktop app with more immersive navigation bars
* Fixed and issue where the mobile PWA app ignored the system navigation bar and notch, which made some UI elements not easy to reach
* On mobile, the player:
  * Shows a play/pause button on the middle of the screen and behaves like the YouTube/Netflix player
  * When double tapping on the right/left side of the screen it skips 10 seconds or goes back 10 seconds instantly
  * Swiping up on the video enter full screen, and swiping down exits full screen
  * There's a new button in the 3 dots menu to open the video stream in nPlay (on iOS only)
  * On iOS, the volume slider is hidden, as there's not way to actually change the volume of a video using a `<video>` tag
  * On iOS. a new Airplay button (needs https://github.com/Stremio/stremio-video/pull/74 to work), and a button in the 3 dots menu to open the vidoe in [nPlayer](https://apps.apple.com/us/app/nplayer-lite/id1078835991)
  * On iOS, on landscape, there's a small padding in the bottom to make sure the video is not rendered under the navigation bar, and added a workaround to turn the navigation bar black on some devices (testing on a physical device is needed to add this workaround to every device, so I've made sure it only works on the devices I could test this on)
* Various CSS bug fixes
* Fixed a bug where the maskable icon was overwritten in the build by the regular icon
* On the player, in the 3 dots menu, added a button to copy the video download link
* Added dev container to the repo to make it easier to start a development environment

Live demo: https://iedustu.github.io/stremio-web/

#### Related pull requests I created for this effort
* https://github.com/Stremio/stremio-video/pull/74
* https://github.com/Stremio/stremio-translations/pull/796

#### Devices and platforms tested with these changes
* macOS:
  * Arc
  * Chrome
  * Firefox
  * Safari
* Windows:
  * Chrome
  * Edge
* iOS:
  * Safari
  * PWA (Clicking on `Add to homescreen` in the share menu in Safari)
  * Chrome
* iPadOS:
  * Safari
  * PWA (Clicking on `Add to homescreen` in the share menu in Safari)
  * Chrome
  * Tested all of these with an external trackpad and without an external trackpad connected
* Android:
  * Chrome
  * PWA

### Screenshots
#### iOS web app
<img src="https://github.com/Stremio/stremio-web/assets/125146135/ddf0d4e5-562c-4218-8ff8-42555c135437" width="350" />
<img src="https://github.com/Stremio/stremio-web/assets/125146135/b760b1aa-0c78-48b2-91fc-a6ec9ec7dd51" width="350" />
<img src="https://github.com/Stremio/stremio-web/assets/125146135/db98af6c-13a0-4c99-8722-816eb9f0c0eb" width="350" />
<img src="https://github.com/Stremio/stremio-web/assets/125146135/8a507cb9-4c6a-4abf-806b-8cf21083a940" width="350" />
<img src="https://github.com/Stremio/stremio-web/assets/125146135/1507b43a-0df0-4f4c-a2be-f66a27f9edf2" width="350" />

#### New mobile player design
<img src="https://github.com/Stremio/stremio-web/assets/125146135/5a6d9415-40ca-452f-a751-554e1f93d0b7" height="350" />
<img src="https://github.com/Stremio/stremio-web/assets/125146135/c1d5e45a-fb16-4d52-8594-0352188eac8f" height="350" />

#### New mobile player design on iOS
<img src="https://github.com/Stremio/stremio-web/assets/125146135/84734eeb-7546-4a4d-9f78-5dddb0354eee" height="350" />
<img src="https://github.com/Stremio/stremio-web/assets/125146135/4f297238-fee9-4a66-966f-4c330084c828" height="350" />

#### Mobile web
<img src="https://github.com/Stremio/stremio-web/assets/125146135/544dfdb0-bb3a-450f-9ab4-e043bc8b21c7" width="350" />
<img src="https://github.com/Stremio/stremio-web/assets/125146135/6a11da0f-c23a-44f2-8b3b-5823d0359410" width="350" />
<img src="https://github.com/Stremio/stremio-web/assets/125146135/5f7e4713-b588-44ab-8c5a-a3fd22ceb57e" width="350" />
<img src="https://github.com/Stremio/stremio-web/assets/125146135/16f3a0bd-cebc-4703-ae70-5c45a9689996" width="350" />
<img src="https://github.com/Stremio/stremio-web/assets/125146135/19346a1b-a3a4-4853-ac8e-6c19729b42d9" width="350" />
<img src="https://github.com/Stremio/stremio-web/assets/125146135/a63949fe-55a5-40b1-b8fb-f8faf747b5d7" width="350" />
<img src="https://github.com/Stremio/stremio-web/assets/125146135/405fa90a-804d-420a-8d2e-90deb76592b0" width="350" />
<img src="https://github.com/Stremio/stremio-web/assets/125146135/3fb73d89-3cb0-4914-be77-bb97842261b5" width="350" />

#### Desktop web
![3-1](https://github.com/Stremio/stremio-web/assets/125146135/68c44b94-7685-4670-959e-3b5c17cac1f7)
![3-2](https://github.com/Stremio/stremio-web/assets/125146135/fa3da6e8-be64-4915-b003-3fbe7971fdb7)
![3-3](https://github.com/Stremio/stremio-web/assets/125146135/822afbc0-204f-42be-8a0a-5a5327ea62a5)
![3-4](https://github.com/Stremio/stremio-web/assets/125146135/9f093813-b6f1-414e-acf5-4c1d263dff32)
![3-5](https://github.com/Stremio/stremio-web/assets/125146135/50e6bb29-b707-4b2e-9b0b-0b6621b78e77)
![3-6](https://github.com/Stremio/stremio-web/assets/125146135/5211e5a5-aad3-4f78-83ea-9c8f29530ce9)
![3-7](https://github.com/Stremio/stremio-web/assets/125146135/3b866b70-c29d-4160-b6bb-fe35218477db)

---

### Screenshots of how it looked before (for reference)
#### iOS web app
<img src="https://github.com/Stremio/stremio-web/assets/125146135/632a1ba6-7ce0-4f90-99df-830bf62ddad0" width="350" />
<img src="https://github.com/Stremio/stremio-web/assets/125146135/860c0807-012a-40bf-b13e-2320beca4cb3" width="350" />
<img src="https://github.com/Stremio/stremio-web/assets/125146135/81236f47-dd93-417d-a683-f253eaaab1a6" width="350" />
<img src="https://github.com/Stremio/stremio-web/assets/125146135/9ac21067-e2fb-49ef-9a9f-440ff53117cf" width="350" />
<img src="https://github.com/Stremio/stremio-web/assets/125146135/54fa440a-8334-42e8-99e3-4644937295f9" width="350" />
<img src="https://github.com/Stremio/stremio-web/assets/125146135/0bd2ccf0-8aa2-4a62-a647-e5e86947fabf" width="350" />
<img src="https://github.com/Stremio/stremio-web/assets/125146135/4129ca17-b70e-479c-8b75-c7aca460f68a" width="350" />
<img src="https://github.com/Stremio/stremio-web/assets/125146135/c8d39943-b33d-4c39-b489-2a9d02a0d563" width="350" />

#### Mobile player
<img src="https://github.com/Stremio/stremio-web/assets/125146135/f26365b6-7f1f-468c-9ec2-5b6df7f0d0b3" height="350" />
<img src="https://github.com/Stremio/stremio-web/assets/125146135/7cdbada8-8ed1-42a6-9b32-63fc1d5c7dfe" height="350" />

#### Desktop web
![old-desktop-1](https://github.com/Stremio/stremio-web/assets/125146135/1b80cf96-a474-4828-8187-f1eb47904ad1)
![old-desktop-2](https://github.com/Stremio/stremio-web/assets/125146135/5ead6e7e-d7ad-4dbf-abb5-df92ee62217a)

IEduStu committed:
feat: add dev container config

IEduStu committed:
fix: dev container config

IEduStu committed:
fix: make exposed dev port public by default

IEduStu committed:
fix: add note about exposing port in dev container

IEduStu committed:
fix: make maskable icons not get overriden during the build

IEduStu committed:
feat: improve mobile design

IEduStu committed:
feat: improve mobile design

IEduStu committed:
feat: mobile controls overlay

IEduStu committed:
fix: bugs

IEduStu committed:
fix: bugs

IEduStu committed:
fix: bugs

IEduStu committed:
feat: open video in nPlayer on iOS

IEduStu committed:
fix: colors

IEduStu committed:
fix: iOS webapp config

IEduStu committed:
fix: remove double scrollbars

IEduStu committed:
fix: preserve existing desktop play behavior

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/Stremio/stremio-web?pullRequest=629) <br/>All committers have signed the CLA.

IEduStu committed:
build: allow manually triggering the CI

IEduStu committed:
test: fix tests

IEduStu committed:
style: lint

IEduStu committed:
build: make deploying to GitHub Pages work

IEduStu committed:
build: fix config

IEduStu committed:
build: make deploying to GitHub Pages work

tymmesyde commented:
Hi, thank you for your PR
As your PR contains a lot of various things, can you please segment your changes in separate PRs?
Like one for fixes or a feature, UI changes, UX changes, etc..
This will make the review easier and there is potentially some changes that we don't want to include in the project

IEduStu commented:
It wouldn't be practical to separate it into multiple PRs as all the changes affect each other and dependant on each other.
For example, the handling of the notch gaps and adaptations for the mobile webapp are intertwined across all the other changes, I'm not sure how can I separate them.

I wrote the long list to detail all the changes and make it easier to CR, but essentially there are 3 main changes:
* Improvements to the mobile webapp to make it feel native
* Improvements to the player on mobile to make it feel similar to YouTube/Netflix
* Infra changes that I made to make it easier for me to develop all of this, so I left it as part of the PR

You can try the live demo of this here: https://iedustu.github.io/stremio-web/

IEduStu committed:
fix: toggle show controls on tap when buffering on mobile

IEduStu committed:
fix: bugs

IEduStu committed:
fix: support toggling video fullscreen on iOS

IEduStu committed:
fix: bugs

tymmesyde commented:
> It wouldn't be practical to separate it into multiple PRs as all the changes affect each other and dependant on each other. For example, the handling of the notch gaps and adaptations for the mobile webapp are intertwined across all the other changes, I'm not sure how can I separate them.
> 
> I wrote the long list to detail all the changes and make it easier to CR, but essentially there are 3 main changes:
> 
>     * Improvements to the mobile webapp to make it feel native
> 
>     * Improvements to the player on mobile to make it feel similar to YouTube/Netflix
> 
>     * Infra changes that I made to make it easier for me to develop all of this, so I left it as part of the PR
> 
> 
> You can try the live demo of this here: https://iedustu.github.io/stremio-web/

I understand that it would not be practical to separate them, we will discuss which changes we want to keep or not before you do that
All your changes are not dependent with each other
As you listed in your comment, you updated the style of the tabs, server warning, etc ..
You also made changes to the player UX by adding a new component for mobile controls, etc...
And also made various fixes
Those can be different PRs and are not related

unclekingpin commented on .github/workflows/build.yml:
```diff
@@ -19,10 +23,12 @@ jobs:
         run: npm run lint
-      - name: Deploy to GitHub Pages
-        uses: peaceiris/actions-gh-pages@v3
+      - name: Upload docs to GitHub Pages
```
"docs"? what docs?
The CI changes looks good thogh, except for the "docs".

IEduStu replied:
My bad, fixed

TRtomasz commented:
on addons page there is shade visible from the top and over first element, it should only add shades once someone starts scrolling
![image](https://github.com/Stremio/stremio-web/assets/40594253/cad25724-f361-4c1f-8960-0b358136865a)
i think firefox does not like this new effets on the discover,  it produces some artifacts
![image](https://github.com/Stremio/stremio-web/assets/40594253/4576d91a-507a-4e90-90cc-6b8328b5c440)

IEduStu committed:
feat: workaround to hide iOS navigation bar in the player on the webapp

IEduStu committed:
pr: update workflow

kKaskak commented:
## Issues found: 
### 1. Keyboard does not open when trying to log in on PWA (iPadOS iPad Pro 11") (sometimes it opens).

![IMG_4546](https://github.com/Stremio/stremio-web/assets/117831817/b963fe3f-29cf-4d9f-b9b9-9b15d575cc41)

### 2. App bugs out a lot while switching routes or simply trying to open the keyboard, to leave the app and enter again.

![IMG_4548](https://github.com/Stremio/stremio-web/assets/117831817/c41fd5a2-23ce-4812-b7d3-a6eb89d674b7)
![IMG_4550](https://github.com/Stremio/stremio-web/assets/117831817/0a6b45b0-0296-47dd-8f71-a8d3c2f45c3e)

### 3. One letter in the redesigned pop-up for the streaming server is only partially visible (padding fix required).

![IMG_4545](https://github.com/Stremio/stremio-web/assets/117831817/d0206fc2-cb95-4ffd-8a73-8231930c430a)

### 4. There is an issue with shades on Firefox (tested on MacBook Pro M1 the newest macOS Firefox v.123.0.1 (64-bit))

https://github.com/Stremio/stremio-web/assets/117831817/6ba83818-303d-4ced-8c8f-74f69570a461


### 5. Play / Resume icon on the bar is missing? Users might be confused why it was removed. 

![IMG_4549](https://github.com/Stremio/stremio-web/assets/117831817/db0a675e-7f52-4ea7-a929-40b9e2f52e2a)


### 6. Search route preview is broken 

<img width="1674" alt="Screenshot 2024-04-17 at 17 24 39" src="https://github.com/Stremio/stremio-web/assets/117831817/fbdeb64e-a6d9-4094-93db-7c3a405c433a">

### 7. Settings page section buttons are broken 

https://github.com/Stremio/stremio-web/assets/117831817/287f7cdd-d2a1-4be1-a731-d93579259265

### 8. Top bar color is broken see before and after photos

*Before*

![image](https://github.com/Stremio/stremio-web/assets/117831817/1824a088-b1ce-4652-9977-33253884525b)

*After  - (here the app broke after trying to swipe up to exit it)*

 ![image](https://github.com/Stremio/stremio-web/assets/117831817/625ba691-683e-49d3-a1a8-f8e6d117f04c)

## Good points: 

1.  Shades look nice, but we need to ensure cross browser support. In some case scenarios items that are object to shades (top & bottom) are almost invisible which we should take into account. 
2. Redesigned `MetaDetails` page looks fresh, tho also in some case scenarios readability might be poor. (it heavily depends on the background image, and we have a different one for each movie / series)

*Please keep in mind that this is just speaking from the user perspective. I did not review any of the code yet, so there was no code quality review done.*
*Next time please split such large PR into multiple as @tymmesyde mentioned already, addressing one issue / new feature at a time. Large PR are way more time-consuming to review.*

IEduStu committed:
pr: fix shade in addons screen

IEduStu committed:
pr: fix Firefix flickers in the Discover screen

IEduStu commented:
@TRtomasz I fixed both your comments, let me know if you still see flickers in Firefox after the fix

IEduStu committed:
pr: fix layouts edge cases, text cropping, login screen, search screen, settings screen

IEduStu commented:
@kKaskak I fixed all your comments except 5.
In the screenshot you attached the pause button is in the middle of the screen, just like in YouTube/Netflix.
This UI looks like that only on mobile devices; I included tablets in this since this is how it looks on YouTube and Netflix apps on my iPad.

Let me know if you still encounter any more issues after my fixes.

kKaskak commented:
> @kKaskak I fixed all your comments except 5. In the screenshot you attached the pause button is in the middle of the screen, just like in YouTube/Netflix. This UI looks like that only on mobile devices; I included tablets in this since this is how it looks on YouTube and Netflix apps on my iPad.
> 
> Let me know if you still encounter any more issues after my fixes.


https://github.com/Stremio/stremio-web/assets/117831817/223ff8e8-7693-4db2-8ffb-6096ec76d19d

This is still not fixed. After clicking an element, it correctly navigates to it, but there is a ref issue and the active element does not change.


https://github.com/Stremio/stremio-web/assets/117831817/0800c775-b070-4e8e-ab21-bc17c607ce8c

IEduStu committed:
feat: add support for the wordaround to hide iOS system navigation bar for an iPad

kKaskak commented:
![image](https://github.com/Stremio/stremio-web/assets/117831817/ac68a393-ae1b-4bac-afdc-b01212db9dbf)
![image](https://github.com/Stremio/stremio-web/assets/117831817/7563c17a-0df4-4924-b29b-f5f36d0a30c9)
![image](https://github.com/Stremio/stremio-web/assets/117831817/a7f88d52-c20a-4698-92a7-c10caaf250b2)

Other than that good job 👍🏼

IEduStu committed:
fix: focus on the correct settings section button

kKaskak commented:
The last one seems to still not focus as it should  but this I believe is a specific page zoom issue

https://github.com/Stremio/stremio-web/assets/117831817/e51aa3b1-686c-4b8c-b93e-beba8b8c5962

IEduStu committed:
fix: gliches in Safari in Settings screen

IEduStu committed:
fix: layout bugs, settings screen bug

IEduStu commented:
@kKaskak Thanks for helping me solve these issues :)
Can you please check again if the issues are solved on your device now?

I couldn't reproduce the issue with the black gap and white line at the top on my iPad, so I made a change that I think may help but I couldn't verify it.
If you can record how you reproduce it it could help me a lot.

IEduStu committed:
refactor: switch from `env()` to css vars

unclekingpin commented on .github/workflows/build.yml:
```diff
@@ -19,10 +23,12 @@ jobs:
         run: npm run lint
-      - name: Deploy to GitHub Pages
-        uses: peaceiris/actions-gh-pages@v3
+      - name: Upload build to GitHub Pages
```
@IEduStu Could you create one more branch out of [mobileImprovements](https://github.com/IEduStu/stremio-web/tree/mobileImprovements) branch in your fork, for me to see how the new CD pipeline will behave?

IEduStu replied:
Done.
You can see the CI of a commit on the forked branch [here](https://github.com/IEduStu/stremio-web/actions/runs/8760374269).
I've make this change to the CI to make it deploy the build to GitHub Pages, since the existing action for this didn't work.

This new code uses GitHub Action's official actions to deploy to GitHub Pages.
To make it work for any fork, you just have to enable GitHub Actions on the fork and enable GitHub Pages in the repo settings and it would work without configuring any environment variable or something like that.

unclekingpin replied:
The expected result is a new directory in the gh-pages branch, with the name of the new branch  so that we can access it at `https://stremio.github.io/stremio-web/<new-branch-name>/`

The goal is to have all branches deployed to gh pages simultaniously

IEduStu replied:
I originally made this change because I thought the deployment didn't work, but it seems that it did just in a different manner than I anticipated.
This is really cool :)

I see how useful it is to have it for a repo with multiple developers working on it, but I think it may lead other forks the same path that I went - thinking the deployment didn't work.
So I made another change for the CI to use the previous action when an environment variable named `GITHUB_PAGES_MULTI_BRANCH` is set to `true`, and otherwise use the simpler deploy action.

This way you can set `GITHUB_PAGES_MULTI_BRANCH` to `true` in this repo settings and preserve the current behavior, and forks that people create won't inherit this environment variable and thus will have the simpler deploy action.
The idea of this is to remove friction in developing this repo on forks, while preserving the current CI behavior.

Let me know if you think it's a good idea or whether should I just revert this whole CI change.

IEduStu committed:
fix: no hover on touch screens

IEduStu committed:
fix: focus scroll on scrollable elements

IEduStu committed:
fix: improve navigation buttons contrast

IEduStu committed:
fix: make slider work correctly on touch screens

IEduStu committed:
fix: make screen switching animation feel smoother

IEduStu committed:
fix: animation was too long in the user menu

IEduStu committed:
fix: fine-tune touch gesture recofnition numbers

IEduStu committed:
fix: smaller UI areas on mobile in the player

IEduStu committed:
fix: subtitles should not render under unsafe areas

IEduStu committed:
fix: double tap gesture called twice

IEduStu committed:
fix: toast shadow was clipped

IEduStu committed:
fix: video area should expand blow unsafe areas

IEduStu committed:
feat: airplay support on Safari

IEduStu committed:
fix: airplay active state effects

IEduStu committed:
fix: improve Airplay active state detection

IEduStu committed:
fix: show the Airplay button on all Safari browsers

kKaskak commented:
Why would we want to disable all hover effects on PWA?  Why would we remove the labels on the `navbar`? Why would we make it harder to actually click on `navbar` buttons?

IEduStu committed:
fix: toasts container caused a scroll on the library screen

IEduStu committed:
fix: make player menu font more clear

IEduStu committed:
fix: don't render embedded subtitles under safe areas

IEduStu committed:
fix: pad top and bottom sides of the video to not cover the bottom unsafe area with subtitles

IEduStu committed:
fix: only pad the bottom of the video

IEduStu commented:
@kKaskak I disabled the hover effects only when you you a "coarse" device, AKA a finger on a touch screen, since without it this leaves the hover state on the button after you click it until you touch something else, which isn't the behavior of native apps.

I haven't removed the labels on the navbar, the functionality is the same as before. What I did do is to increase the text contrast a bit when the navbar is at the bottom without showing the navbar labels when the navbar is on the left of the screen (like it was before).

I didn't make it any harder to click navbar buttons; the effect of the clicks is now even faster than it was before which makes it easier to navigate the app.

Try [the latest build](https://iedustu.github.io/stremio-web/) and see firsthand how it looks and behaves.
This URL updates with every change I make, you can install it as a PWA and see it in action.

kKaskak commented:
> @kKaskak I disabled the hover effects only when you you a "coarse" device, AKA a finger on a touch screen, since without it this leaves the hover state on the button after you click it until you touch something else, which isn't the behavior of native apps.
> 
> I haven't removed the labels on the navbar, the functionality is the same as before. What I did do is to increase the text contrast a bit when the navbar is at the bottom without showing the navbar labels when the navbar is on the left of the screen (like it was before).
> 
> I didn't make it any harder to click navbar buttons; the effect of the clicks is now even faster than it was before which makes it easier to navigate the app.
> 
> Try [the latest build](https://iedustu.github.io/stremio-web/) and see firsthand how it looks and behaves. This URL updates with every change I make, you can install it as a PWA and see it in action.

Could you explain in more detail why the decision to lower accessibility on PWA by removing all hover effects would benefit our project? Also labels are there for a reason when you tap / hover the navbar buttons.

IEduStu committed:
pr: support multi-branch and single-branch GitHub Pages deployments

IEduStu commented:
@kKaskak When you move a mouse over an element it triggers `:hover`, when you tap an element on a touchscreen it also triggeres `:hover` and it stays like that until you tap outside the element.
The handling of `:hover` for a mouse and a finger on a touchscreen should be different, since on desktop you need to have an effect when a mouse is over a button for example, but on a touch screen there's no really such a thing as a hover - you can either not touch the screen or do touch the screen (in which case you'd trigger `:active` for as long as the finger is down on the screen, but unlike `:hover`, when you lift the finger from the screen then `:active` will end).

`:hover` works the way it does on touchscreens since there are some websites that when you hover over a button it can open a submenu, using only CSS, and emulating `:hover` is required for this to actually work on mobile.
Usually, when you define `:hover` for an element you intend to only change its design when the user points the mouse over it, but not trigger for what technically `:hover` does on mobile.

On native mobile apps (for example), after you click a button, there's no effect left on it after you lift your finger off the screen, so a PWA used on a touchscreen should behave the same.

In the CSS code that I changed, you can see that I wrapped only `:hover`s with a media query like this:
```css
@media (pointer: fine) {
    .my-class:hover {
        background-color: rgba(0, 0, 0, 0.24);
    }
}
```

A [`coarse` pointer is a touchscreen](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/pointer#coarse), and a [`fine` pointer is a mouse](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/pointer#fine) (for this case), so this CSS code makes sure that only on devices where the primary input device is a mouse `:hover` will work.

This CSS code is meant to preserve the current behavior for desktop devices and make the behavior of elements on mobile devices more intiuitive and native-like.

The only case where it may feel different than what you're used to is when you use a mouse on a touchscreen device where mouse is not the primary input device, like an iPad, in which case the hover effects will no longer be visible even when you use a mouse.
If you'd like to enable the hover effect on touchscreens when a mouse is connected to the device (like when using a mouse or a touchpad with an iPad), it's possible to do by changing the media query to `@media (any-hover: hover) { ... }`, but this may look more like a bug when the user uses their finger and not their mouse.
Let me know if you'd like it to work like that and I'll make the change.

I'm not sure what you mean about lowering the accessibility of the project, as I haven't made any change that hides visible information, and focus styles are still in place just like before.

IEduStu committed:
fix: slider touch end event

IEduStu committed:
fix: remove duplicate top navbar top safe inset margin

kKaskak commented:
> @kKaskak When you move a mouse over an element it triggers `:hover`, when you tap an element on a touchscreen it also triggeres `:hover` and it stays like that until you tap outside the element. The handling of `:hover` for a mouse and a finger on a touchscreen should be different, since on desktop you need to have an effect when a mouse is over a button for example, but on a touch screen there's no really such a thing as a hover - you can either not touch the screen or do touch the screen (in which case you'd trigger `:active` for as long as the finger is down on the screen, but unlike `:hover`, when you lift the finger from the screen then `:active` will end).
> 
> `:hover` works the way it does on touchscreens since there are some websites that when you hover over a button it can open a submenu, using only CSS, and emulating `:hover` is required for this to actually work on mobile. Usually, when you define `:hover` for an element you intend to only change its design when the user points the mouse over it, but not trigger for what technically `:hover` does on mobile.
> 
> On native mobile apps (for example), after you click a button, there's no effect left on it after you lift your finger off the screen, so a PWA used on a touchscreen should behave the same.
> 
> In the CSS code that I changed, you can see that I wrapped only `:hover`s with a media query like this:
> 
> ```css
> @media (pointer: fine) {
>     .my-class:hover {
>         background-color: rgba(0, 0, 0, 0.24);
>     }
> }
> ```
> 
> A [`coarse` pointer is a touchscreen](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/pointer#coarse), and a [`fine` pointer is a mouse](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/pointer#fine) (for this case), so this CSS code makes sure that only on devices where the primary input device is a mouse `:hover` will work.
> 
> This CSS code is meant to preserve the current behavior for desktop devices and make the behavior of elements on mobile devices more intiuitive and native-like.
> 
> The only case where it may feel different than what you're used to is when you use a mouse on a touchscreen device where mouse is not the primary input device, like an iPad, in which case the hover effects will no longer be visible even when you use a mouse. If you'd like to enable the hover effect on touchscreens when a mouse is connected to the device (like when using a mouse or a touchpad with an iPad), it's possible to do by changing the media query to `@media (any-hover: hover) { ... }`, but this may look more like a bug when the user uses their finger and not their mouse. Let me know if you'd like it to work like that and I'll make the change.
> 
> I'm not sure what you mean about lowering the accessibility of the project, as I haven't made any change that hides visible information, and focus styles are still in place just like before.

We'll discuss this with the team and I will give you a final answer.

IEduStu committed:
fix: make sure the scrollbar is not cropped on all platforms, improve immersiveness

IEduStu committed:
fix: generic detection of overlay scrollbars

IEduStu committed:
fix: when clicking on the `Download this video` button in the player, use the original filename for the downloaded file instead of `2`

IEduStu committed:
fix: simplify download video button logic

IEduStu committed:
feat: copy download link button, use i18n

IEduStu committed:
fix: align texts with new translations PR

IEduStu committed:
fix: workarounds for CSS issues on Safari

IEduStu committed:
fix: position issue

IEduStu committed:
fix: check for `overflow: clip` support

IEduStu commented:
I've updated the screenshots to the latest version of this branch and detailed all the platforms and browsers I've tested this on to make sure everything works correctly.

If you still find issues, let me know so I can fix them, but I think that the current version of the PR is stable and covers all the platforms and browsers I could test this on.
I've been using it daily for a while now to spot all the issues I could find and I think I got them all.

I've attempted to maximixe the capbilities of Safari on iOS and iPadOS, and Chrome on Android, to make the PWA feel as native as possible on mobile devices, while making sure the desktop version also works great.

kKaskak commented:
I started to extensively test your PR on my android device as a PWA. One thing I noticed is that local search no longer appears under search history. I will be testing it by simply using, the coming days. Be patient as I might be slow to review everything. Have a nice weekend

kKaskak commented:
Hi. I noticed a small offset issue in the back button in the player route. Can you fix that?

kKaskak commented:
https://github.com/Stremio/stremio-web/assets/117831817/24a694e7-8f12-4c78-925b-515a9935d190

IEduStu committed:
Merge remote-tracking branch 'origin/development' into mobileImprovements

IEduStu commented:
@kKaskak I couldn't reproduce this issue where the back button jumps up.
How can I get to a situation where it happens?

kKaskak commented:
> @kKaskak I couldn't reproduce this issue where the back button jumps up.
> 
> How can I get to a situation where it happens?

As I previously mentioned when you enter the player it's not in the same position as on other routes.

IEduStu committed:
fix: shrink header in player only when there's no enough space

IEduStu commented:
I see what you mean.
On desktop it's on the same place, but on mobile I've shrunk the height of the header a bit to make more space available to touch the video area to hide the play/pause overlay.
This was very beneficial for some smarphone screen sizes I tested this on, but is irrelevant when the screen is tall enough, so I've added a fix to only shrink the header this way when the screen height is not tall enough.

kKaskak commented:
There is one more thing that should be adjusted. The top navigation bar in iOS PWA now has a default white color which isn't the case on `/development` 
![image](https://github.com/Stremio/stremio-web/assets/117831817/aa7b9280-0dd1-4368-9b35-5a51a3a32686)
![image](https://github.com/Stremio/stremio-web/assets/117831817/697007bb-a1b8-4e43-beb6-9c594546bc9c)

IEduStu commented:
I've just installed the PWA app from [this link](https://iedustu.github.io/stremio-web/) on my iPad and there's no white bar at the top.
The white bar was an issue with earlier versions of this PR, so if you installed the PWA with an older version you have to reinstall the PWA it fo fix it.

gazer000 commented:
On the ModalDialog styles, the `.buttons-container` custom style for mobile screen should be nested under `.modal-dialog-content` otherwise the modal's button list flex direction stays `row` and texts will clip on mobile screen

kKaskak commented:
> I've just installed the PWA app from [this link](https://iedustu.github.io/stremio-web/) on my iPad and there's no white bar at the top.
> The white bar was an issue with earlier versions of this PR, so if you installed the PWA with an older version you have to reinstall the PWA it fo fix it.

I'm sure I'm running the latest version, and it still happens.
I also installed a new version to be sure 100%, it still shows up.

kKaskak commented:
> On the ModalDialog styles, the `.buttons-container` custom style for mobile screen should be nested under `.modal-dialog-content` otherwise the modal's button list flex direction stays `row` and texts will clip on mobile screen

Yes that's true. Thanks for mentioning it.

IEduStu committed:
fix: dialog buttons direction on mobile

IEduStu committed:
fix: attempt to fix status bar color issue

IEduStu commented:
@kKaskak Can you please share with me what version of iPadOS do you have installed and whether there are custom accessibility features enabled on your iPad?
I couldn't reproduce the issue with the status bar color that you experience, but I've made some change that might fix it, althougt I cannot test that it does something on my iPad.

Essentially, this PR changes the `meta` tag in the `head` of the HTML to:
```html
<meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
```
Which according to [Apple's documentation](https://developer.apple.com/library/archive/documentation/AppleApplications/Reference/SafariHTMLRef/Articles/MetaTags.html) makes the status bar transparent.
Then, I made sure to cover the area of the status bar with CSS to make the gradient of the background of the app go under the status bar.

This is how it looks on my iPad Pro 12.9" with iPadOS 17.4.1:
![1](https://github.com/Stremio/stremio-web/assets/125146135/467acb46-98fc-41aa-89fd-dbdcf372c6aa)
![2](https://github.com/Stremio/stremio-web/assets/125146135/fce65692-1989-4035-9332-022ddacaabbb)

BTW I also fixed the issue with `.buttons-container` in `ModalDialog` on mobile

**Update:** I just tested it again with my iPad updated to iPadOS 17.5.1 and I still couldn't reproduce the issue with the status bar color

kKaskak commented:
It works fine now on iPad Pro iPadOS 17.4.1 
Thank you for the change. Can you also ensure there are no conflicts between your branch and the development branch? 🚀

kKaskak commented:
Buttons issue seems to be resolved too 🔥

IEduStu commented:
@kKaskak It seems that my branch is up to date against the latest development branch and there are no conflicts

IEduStu commented:
@kKaskak Do you plan on merging this PR soon?
It keeps accruing conflicts as the main branch gets more features and I don't want to continue resolving conflicts if you don't plan to merge it

kKaskak commented:
> @kKaskak Do you plan on merging this PR soon? It keeps accruing conflicts as the main branch gets more features and I don't want to continue resolving conflicts if you don't plan to merge it

Please resolve the last conflict and it might be merged soon.

Viren070 commented:
On the settings page, when you click one of the shortcuts (General, Player, Streaming, Shortcuts) it scrolls down but that section isn't highlighted. I tested this on https://iedustu.github.io/stremio-web/ on Windows 11 running Brave. 


https://github.com/user-attachments/assets/82a36141-569b-446a-a303-e86e45e7f553


It highlights correctly on https://stremio.github.io/stremio-web/development and https://web.stremio.com

kKaskak commented:
> On the settings page, when you click one of the shortcuts (General, Player, Streaming, Shortcuts) it scrolls down but that section isn't highlighted. I tested this on https://iedustu.github.io/stremio-web/ on Windows 11 running Brave.
> 
>  Screen.Recording.2024-07-19.170304.mp4 
> It highlights correctly on https://stremio.github.io/stremio-web/development and https://web.stremio.com

Good catch, this still needs to be fixed, as I mentioned before!
https://github.com/Stremio/stremio-web/pull/629#issuecomment-2061376856

IEduStu committed:
Merge remote-tracking branch 'origin/development' into mobileImprovements

IEduStu committed:
fix: focused settings scrtion buttons on some zoom levels

IEduStu commented:
> > On the settings page, when you click one of the shortcuts (General, Player, Streaming, Shortcuts) it scrolls down but that section isn't highlighted. I tested this on https://iedustu.github.io/stremio-web/ on Windows 11 running Brave.
> > Screen.Recording.2024-07-19.170304.mp4
> > It highlights correctly on https://stremio.github.io/stremio-web/development and https://web.stremio.com
> 
> Good catch, this still needs to be fixed, as I mentioned before! [#629 (comment)](https://github.com/Stremio/stremio-web/pull/629#issuecomment-2061376856)

I indeed already fixed this issue, but there it seems that it was still there when you use the browser's zoom (via `cmd`+`+`), and is also happens on the development branch (although it happens with different zoom levels than this branch).
In any case, I fixed this issue so it now works as expected also when you zoom.

IEduStu committed:
chore: use new translation texts from sub-PR

IEduStu commented:
> > @kKaskak Do you plan on merging this PR soon? It keeps accruing conflicts as the main branch gets more features and I don't want to continue resolving conflicts if you don't plan to merge it
> 
> Please resolve the last conflict and it might be merged soon.

I fixed all the conflicts and issues, I hope you can merge it soon.

Viren070 commented:
![image](https://github.com/user-attachments/assets/2bbb3477-86a1-4481-a4e0-f6328e9ccb76)

Regarding the streaming service banner: 
There's padding between the edge of the screen at the sides but not at the bottom. Personally, I believe it would look better if there was also some padding at the bottom or no padding at all or remove the corner radius for the bottom part. 

Also, could I ask if the fix for the settings section button is live at https://iedustu.github.io/stremio-web/ ? They still don't highlight correctly and I'm on default (100%) zoom. I do see that some workflows failed to run for your latest commits.

kKaskak commented:
Could you also address the changes that @unclekingpin requested? If they are already addressed please mark as resolved. Thank you

kKaskak commented:
> > > On the settings page, when you click one of the shortcuts (General, Player, Streaming, Shortcuts) it scrolls down but that section isn't highlighted. I tested this on https://iedustu.github.io/stremio-web/ on Windows 11 running Brave.
> > > Screen.Recording.2024-07-19.170304.mp4
> > > It highlights correctly on https://stremio.github.io/stremio-web/development and https://web.stremio.com
> > 
> > 
> > Good catch, this still needs to be fixed, as I mentioned before! [#629 (comment)](https://github.com/Stremio/stremio-web/pull/629#issuecomment-2061376856)
> 
> I indeed already fixed this issue, but there it seems that it was still there when you use the browser's zoom (via `cmd`+`+`), and is also happens on the development branch (although it happens with different zoom levels than this branch). In any case, I fixed this issue so it now works as expected also when you zoom.

Still doesn't work as expected in my case, running Chrome on Ubuntu with 100% zoom.
[Screencast from 2024-07-25 11-23-51.webm](https://github.com/user-attachments/assets/824c81ba-e1b7-4cbe-9f72-bd51519b3e9a)

I tested on  https://iedustu.github.io/stremio-web/

IEduStu committed:
fix: streaming server message margin

IEduStu committed:
fix: CI workflow

IEduStu commented:
> Regarding the streaming service banner [...]
> Personally, I believe it would look better if there was also some padding at the bottom

Fixed.


> Also, could I ask if the fix for the settings section button is live at https://iedustu.github.io/stremio-web/ ? They still don't highlight correctly and I'm on default (100%) zoom. I do see that some workflows failed to run for your latest commits

You're right, I missed that.
I fixed the CI now and the fix is now live at  https://iedustu.github.io/stremio-web/


> Could you also address the changes that @unclekingpin requested? If they are already addressed please mark as resolved. Thank you

Done.
Before you merge this PR, make sure to go to the project settings in GitHub > Secrets and variables > Actions > variables tab > Repository variables > New repository variable, and then add a variable called `GITHUB_PAGES_MULTI_BRANCH` with its value set to `true`.
This is needed to preserve the existing method used for deployment to GitHub Pages that you currently have, while making it much easier for forks to deploy to GitHub Pages (like I did for this PR)


> Still doesn't work as expected in my case, running Chrome on Ubuntu with 100% zoom

Try again now, the fix should be live now.
Also, try reloading with `cmd`+`shift`+`R`, since it seems that a cache of the page before the fix was used on my device so I had to refresh this way to flush it to get the fix.

Viren070 commented:
Thanks for fixing the banner. 

However, the section buttons still don't highlight correctly (still Brave on Windows 11). 

Brave (100% zoom) on Windows 11 (100% scaling) gives me this: 

https://github.com/user-attachments/assets/caf0313c-dc36-43e6-9862-4de0c5b941f2

However, if I set the browser zoom to 125% or 150% it does highlight correctly.

If I change the window size to something slightly smaller, it also highlights the correct section. Here is a video showing what I mean: 

https://github.com/user-attachments/assets/d23a3cb5-ecdf-4337-9a35-430acaf80e0e

If you need more information please let me know. I am looking forward to this being merged.

IEduStu committed:
fix: focused settings section buttons

IEduStu commented:
@Viren070 I couldn't reproduce the issue now on my machine, so I've applied a fix that I think may solve this.
Can you please check now?

Also, after clicking on a section button, if the button is not focused, is scrolling down 1 pixel makes the button be focused?

If the issue still happens, can you please run this script and share with me its result?
Run this in the devtools console after clicking a section button and having the button not being focused.
```javascript
console.log("innerWidth", window.innerWidth);
console.log("outerWidth", window.outerWidth);
console.log("innerHeight", window.innerHeight);
console.log("outerHeight", window.outerHeight);
console.log("devicePixelRatio", window.devicePixelRatio);
console.log("scrollTop", document.querySelector('[class^="sections-container"]')?.scrollTop);
console.log("sections", JSON.stringify(
    Array.from(document.querySelector('[class^="sections-container"]').children ?? [])
        .map((item) => ({
            className: item.className,
            offsetTop: item.offsetTop
        }))
    )
);
console.log("scrollPaddingTop", getComputedStyle(document.querySelector('[class^="sections-container"]')).scrollPaddingTop);
```

Viren070 commented:
> @Viren070 I couldn't reproduce the issue now on my machine, so I've applied a fix that I think may solve this. Can you please check now?

I have tried this again and the issue still occurs 

> Also, after clicking on a section button, if the button is not focused, is scrolling down 1 pixel makes the button be focused?

Yes. 

> If the issue still happens, can you please run this script and share with me its result? Run this in the devtools console after clicking a section button and having the button not being focused.
> 
> ```js
> console.log("innerWidth", window.innerWidth);
> console.log("outerWidth", window.outerWidth);
> console.log("innerHeight", window.innerHeight);
> console.log("outerHeight", window.outerHeight);
> console.log("devicePixelRatio", window.devicePixelRatio);
> console.log("scrollTop", document.querySelector('[class^="sections-container"]')?.scrollTop);
> console.log("sections", JSON.stringify(
>     Array.from(document.querySelector('[class^="sections-container"]').children ?? [])
>         .map((item) => ({
>             className: item.className,
>             offsetTop: item.offsetTop
>         }))
>     )
> );
> console.log("scrollPaddingTop", getComputedStyle(document.querySelector('[class^="sections-container"]')).scrollPaddingTop);
> ```
After running this, I got the following output: 

```
innerWidth 837
outerWidth 1920
innerHeight 951
outerHeight 1032
devicePixelRatio 1
scrollTop 708
sections [{"className":"section-container-twzKQ","offsetTop":77},{"className":"section-container-twzKQ","offsetTop":309},{"className":"section-container-twzKQ","offsetTop":604},{"className":"section-container-twzKQ","offsetTop":738},{"className":"section-container-twzKQ","offsetTop":1324},{"className":"section-container-twzKQ","offsetTop":1593},{"className":"section-container-twzKQ","offsetTop":1959},{"className":"section-container-twzKQ","offsetTop":2227},{"className":"section-container-twzKQ","offsetTop":2514},{"className":"section-container-twzKQ","offsetTop":2892},{"className":"section-container-twzKQ versions-section-container-LRCVQ","offsetTop":3962}]
scrollPaddingTop 77px
```

IEduStu committed:
fix: focused settings section buttons

IEduStu commented:
@Viren070 I think I found the issue and fixed it.
Can you please check again?

Viren070 commented:
@IEduStu It works!

IEduStu committed:
Merge remote-tracking branch 'origin/development' into mobileImprovements

IEduStu commented:
I just had to resolve another conflict now :/

kKaskak commented on src/App/styles.less:
```diff
@@ -40,6 +40,8 @@
     --outer-glow: 0px 0px 30px rgba(123, 91, 245, 0.37);
     --border-radius: 0.75rem;
+
+    background-color: #161523;
```
Can you use one of the existing variables instead of creating a new one?

kKaskak replied:
If I understood correctly, this used to change the  "apple-mobile-web-app-status-bar" style?

IEduStu replied:
I copied this value from the `WebpackPwaManifest` config in `webpack.config.js`, so since it has to be hard-coded there, I thought it'd make sense to do the same here.
I've added a comment about that to make it more clear.

This ensures the title bar color is consistence across devices and browsers, instead of fallbacking to each browser's behavior which are inconsistent.

kKaskak commented on src/common/useFullscreen.js:
```
// Copyright (C) 2017-2023 Smart code 203358507

const React = require('react');

const useFullscreen = (videoElementRef) => {
    const getVideoElement = React.useCallback(() => {
        if (videoElementRef && videoElementRef.current) {
            if (videoElementRef.current instanceof HTMLVideoElement)
                return videoElementRef.current;
            else
                return videoElementRef.current.querySelector('video');
        }

        return null;
    }, [videoElementRef]);

    const [fullscreen, setFullscreen] = React.useState(false);

    const requestFullscreen = React.useCallback(() => {
        if (document.documentElement.requestFullscreen) {
            document.documentElement.requestFullscreen();
        } else {
            const videoElement = getVideoElement();
            if (videoElement && videoElement.webkitEnterFullscreen) {
                videoElement.webkitEnterFullscreen();
            }
        }
    }, [getVideoElement]);
    const exitFullscreen = React.useCallback(() => {
        if (document.exitFullscreen) {
            document.exitFullscreen();
        } else {
            const videoElement = getVideoElement();
            if (videoElement && videoElement.webkitExitFullscreen) {
                videoElement.webkitExitFullscreen();
            }
        }
    }, [getVideoElement]);
    const toggleFullscreen = React.useCallback(() => {
        fullscreen ? exitFullscreen() : requestFullscreen();
    }, [fullscreen, exitFullscreen, requestFullscreen]);
    React.useEffect(() => {
        const videoElement = getVideoElement();

        const onFullscreenChange = () => {
            setFullscreen(document.fullscreenElement === document.documentElement || (videoElement && videoElement.webkitDisplayingFullscreen));
        };

        document.addEventListener('fullscreenchange', onFullscreenChange);
        if (videoElement && videoElement.addEventListener) {
            videoElement.addEventListener('webkitenterfullscreen', onFullscreenChange);
            videoElement.addEventListener('webkitendfullscreen', onFullscreenChange);
        }

        return () => {
            document.removeEventListener('fullscreenchange', onFullscreenChange);

            if (videoElement && videoElement.removeEventListener) {
                videoElement.removeEventListener('webkitenterfullscreen', onFullscreenChange);
                videoElement.removeEventListener('webkitendfullscreen', onFullscreenChange);
            }
        };
    }, []);

    return [fullscreen, requestFullscreen, exitFullscreen, toggleFullscreen];
};

module.exports = useFullscreen;

```

kKaskak replied:
Reduce redundant calls during video playback and optimize performance.

IEduStu replied:
Done

kKaskak commented on src/routes/Addons/styles.less:
Here you can use variables like this at the start of the file:
```
@selectable-inputs-assumed-height: 84px;
@top-overlay-size: @@selectable-inputs-assumed-height;
@bottom-vertical-nav-bar-size: 0px;
@bottom-overlay-size: var(@bottom-vertical-nav-bar-size);
@overlap-size: 6rem;
@transparency-grandient-pad: 3rem;
```

kKaskak replied:
This will clear up the code of all the vars you will just use 
```
@variable
```

IEduStu replied:
I used CSS variables here since I change them on some scopes based on media queries and other parameters, and it affects all the other dependant calculations.
This is impossible to do using preprocessor variables.

IEduStu replied:
Some of the variables are static, though, so I can extract some of them to the top of the file using preprocessor variables like you suggested, but I think it would make it more cumbersome to read when some of the variables are CSS variables and some are preprocessor variables.

Should I change only the static ones to be preprocessor variables?

kKaskak commented on src/routes/Discover/styles.less:
```diff
@@ -24,13 +24,20 @@
     background-color: transparent;
 
     .discover-content {
+        --selectable-inputs-assumed-height: 84px;
```
You can do the same here too, use `@variable` at the start of the file and define a value for it.
Then reuse as
```
@variable
```

IEduStu replied:
Same issue here with some variables having to be CSS variables.
Let me know what you think I should do in the other comment (https://github.com/Stremio/stremio-web/pull/629#discussion_r1762131566).

kKaskak commented on src/routes/MetaDetails/MetaDetails.js:
```diff
@@ -76,8 +76,23 @@ const MetaDetails = ({ urlParams, queryParams }) => {
     return (
         <div className={styles['metadetails-container']}>
+            {
+                canRenderBackground &&
```
Here you can use the app overall syntax using ` ? : null`

IEduStu replied:
Done

kKaskak commented on src/routes/MetaDetails/styles.less:
```diff
@@ -17,6 +17,34 @@
+    padding-right: var(--safe-area-inset-right, 0px);
+    box-sizing: border-box;
+
+    --navbar-assumed-height: 84px;
```
same here

IEduStu replied:
Here the variables are indeed all static, but I think that for consistency with the other files it may make sense to keep them css vars.
Let me know if you agree or want me to change it to preprocessor variables anyway.

IEduStu committed:
Merge remote-tracking branch 'origin/development' into mobileImprovements

IEduStu commented:
@kKaskak Do you have any estimation of when will this get merged?
I just had to resolve another conflict...

IEduStu commented:
@kKaskak It seems that using a streaming server stopped working on this branch and also on the development branch (via https://stremio.github.io/stremio-web/development).
I couldn't find any recent changes that might have caused this on the development branch.

Do you have an idea of why it happens?
The main website (https://web.stremio.com) seems to work fine though.

kKaskak commented:
> @kKaskak It seems that using a streaming server stopped working on this branch and also on the development branch (via https://stremio.github.io/stremio-web/development).
> 
> I couldn't find any recent changes that might have caused this on the development branch.
> 
> 
> 
> Do you have an idea of why it happens?
> 
> The main website (https://web.stremio.com) seems to work fine though.

It should be working the same as it did before

tsaridas commented:
> @kKaskak It seems that using a streaming server stopped working on this branch and also on the development branch (via https://stremio.github.io/stremio-web/development). I couldn't find any recent changes that might have caused this on the development branch.
> 
> Do you have an idea of why it happens? The main website (https://web.stremio.com) seems to work fine though.

try adding 
```
"proxyStreamsEnabled": false
```
to your server-settings.json

IEduStu committed:
fix: CR comments

IEduStu committed:
style: lint

IEduStu committed:
Merge remote-tracking branch 'origin/development' into mobileImprovements

IEduStu committed:
fix: rever to last not broken `@stremio/stremio-core-web` version

IEduStu commented:
> > @kKaskak It seems that using a streaming server stopped working on this branch and also on the development branch (via https://stremio.github.io/stremio-web/development).
> > I couldn't find any recent changes that might have caused this on the development branch.
> > Do you have an idea of why it happens?
> > The main website (https://web.stremio.com) seems to work fine though.
> 
> It should be working the same as it did before

I found the cause for this - the latest version of `@stremio/stremio-core-web` (0.47.7) is broken and causes the connection to the streaming server to fail.
I reverted it to `0.47.6` on this PR and it now works.
Try using the [development branch](https://stremio.github.io/stremio-web/development) and notice that if fails to connect to the streaming server, while [this PR](https://iedustu.github.io/stremio-web/) works (refresh the page opening it, since the cache from previous load it used by default).

This seems to be the commit that caused this issue: https://github.com/Stremio/stremio-web/commit/69a1a8b92e50c302f5699f61671fa2e311cd5127

IEduStu committed:
Merge remote-tracking branch 'origin/development' into mobileImprovements

IEduStu commented:
@kKaskak I see that you now fixed the issue with `@stremio/stremio-core-web` that I mentioned.
_update_: I see that @elpiel fixed it. Thanks @elpiel :)

This PR just accrued another conflict that I had to solve...
This PR is open for months now, I fixed all the bugs a long time ago, and since then it just accrued a conflict after conflict, which I solved every time like you asked so you can finally merge it.

At this point, I think you can just merge it and make all the changes you want afterwards, since small formatting changes shouldn't be the reason to wait for another 2 months before reviewing this PR again, which by then I'll have to solve many more conflicts.

I don't mean to be rude, but if you don't plan to merge this then please just let me know so I can stop wasting time resolving conflicts...

IEduStu committed:
Merge remote-tracking branch 'origin/development' into mobileImprovements

tymmesyde commented:
@IEduStu Hi, we need you to split your changes into multiple PRs so we can review and merge them
Can you start by making a PR for the mobile player changes?

IEduStu commented:
@tymmesyde The mobile player depends on the style changes that make the app feel more native, since it goes to great lengths to ensure that the entire screen is used when the website is installed as PWA (including the status bar and the navigation bar areas), and the mobile player heavily depends on that.
I can extract a PR with only the changes to the style of the app without the mobile player first, and after that is merged, then open a new PR with the mobile player changes.
Does it sound good?

I'd also like to have some sort of estimation of a timeline for that to be reviewed and merged since I don't want to work again on something and end up waiting again for another 5 months in which I have to maintain it and merge conflicts with the main branch all the time.

IEduStu committed:
fix: resolve relative `streamingServerUrl` query param

kKaskak commented:
I will add all of the PR with changes that were cherrypicked from this PR in this comment for future reference: 
- https://github.com/Stremio/stremio-web/pull/784
- https://github.com/Stremio/stremio-web/pull/765
- https://github.com/Stremio/stremio-web/pull/785
- https://github.com/Stremio/stremio-web/pull/786
- https://github.com/Stremio/stremio-web/pull/796
- https://github.com/Stremio/stremio-web/pull/797

kKaskak commented:
this is all that the designer team approved for now, closing the PR because of clean up
if anything changes i will link up more PRs with cherry picked changes
